kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/ConflictChecker.java

```diff
@@ -57,10 +64,18 @@ public class ConflictChecker {
   private final TransactionImpl transaction;
   private final long attemptVersion;
 
-  private ConflictChecker(SnapshotImpl snapshot, TransactionImpl transaction, long attemptVersion) {
+  // Data actions from the losing transaction
+  private final CloseableIterable<Row> dataActions;
+
+  private ConflictChecker(
+      SnapshotImpl snapshot,
+      TransactionImpl transaction,
+      long attemptVersion,
+      CloseableIterable<Row> dataActions) {
```

Comment:
why are you passing the dataActions here?

Reply:
Previously, domain metadata actions were included within `dataActions`, and they are needed for conflict resolution. In the latest update, I’ve separated domain metadata actions out and they can be accessed from `this.transaction`, so `dataActions` is no longer being passed here.